### DISCUSSION TITLE
Return a `ShutdownScript` when using `ShutdownScript.new_witness_program` (Kotlin/Java)

### DISCUSSION THREAD

**Discussion by ConorOkus** [lightningdevkit/rust-lightning#2129] [Help] [answered]
I'm trying to override LDK's `get_shutdown_scriptpubkey` on the `SignerProvider` in Kotlin. Some work has been done on the BDK side to extract the payload from an address in order to extract and reuse the `WitnessProgram` [here](https://github.com/bitcoindevkit/bdk-ffi/pull/325). 

```kotlin
interface LDKSignerProvider : SignerProvider.SignerProviderInterface {
    override fun get_shutdown_scriptpubkey(): ShutdownScript {
        val address = Address(OnchainWallet.getNewAddress())
        val payload: Payload = address.payload()

        when (payload) {
            is Payload.WitnessProgram -> {
                return ShutdownScript.new_witness_program(
                    WitnessVersion(payload.version.name.toByte()),
                    payload.program.toUByteArray().toByteArray()
                )
            }
            else -> {
                return this._shutdown_scriptpubkey
            }
        }
    }

```

Is it possible for `ShutdownScript.new_witness_program` to return a `ShutdownScript` like `ShutdownScript.new_p2wsh` and `ShutdownScript new_p2wpkh` instead of `Result_ShutdownScriptInvalidShutdownScriptZ`?

**Comment by wpaulino** [comment, marked as answer]
It returns `Result_ShutdownScriptInvalidShutdownScriptZ` because it can fail. I'm not familiar with Kotlin, but you'd need to check if the call failed or not to get a `ShutdownScript` out of it.

**Reply by ConorOkus** [reply]
Yeah it doesn't look like we expose a way to instantiate a new `ShutdownScript` [here](https://github.com/lightningdevkit/ldk-garbagecollected/blob/main/src/main/java/org/ldk/structs/ShutdownScript.java#L17). Am I missing something? @tnull @TheBlueMatt ?

**Reply by tnull** [reply]
IIUC, you should be able to do something like:
```kotlin
val script = (result as Result_ShutdownScriptInvalidShutdownScriptZ.Result_ShutdownScriptInvalidShutdownScriptZ_OK).res
```
or similar (excuse my Kotlin).

**Reply by ConorOkus** [reply]
That worked @tnull, thanks
